packages/runtime/runtime-definitions/src/dataStoreRegistry.ts

```diff
@@ -20,11 +20,14 @@ export type FluidDataStoreRegistryEntry =
  * An associated pair of an identifier and registry entry.  Registry entries
  * may be dynamically loaded.
  */
-export type NamedFluidDataStoreRegistryEntry = [string, Promise<FluidDataStoreRegistryEntry>];
+export type NamedFluidDataStoreRegistryEntry =
+    [string, Promise<FluidDataStoreRegistryEntry> | FluidDataStoreRegistryEntry];
```

Comment:
if it might be a promise why not just always have a promise? what is gained by only sometimes being a promise?